### ISSUE TITLE
[Bug]: ValueError: "HuggingFaceLLM" object has no field "_messages_to_prompt"

### ISSUE THREAD

**Issue by joergwa** [run-llama/llama_index#9458]
### Bug Description

In llama_index/llms/huggingface.py line 132 has been removed.
```
_messages_to_prompt: Callable = PrivateAttr() 
```
Leading to:
```
ValueError: "HuggingFaceLLM" object has no field "_messages_to_prompt"
```

### Version

v0.9.14.post1

### Steps to Reproduce

```
from llama_index.llms import HuggingFaceLLM
llm = HuggingFaceLLM(model_name="mistralai/Mistral-7B-Instruct-v0.1", tokenizer_name="mistralai/Mistral-7B-Instruct-v0.1", device_map="auto")
```
Result:
```
Loading checkpoint shards: 100%
2/2 [00:06<00:00, 2.90s/it]

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[5], line 2
      1 from llama_index.llms import HuggingFaceLLM
----> 2 llm = HuggingFaceLLM(model_name="mistralai/Mistral-7B-Instruct-v0.1", tokenizer_name="mistralai/Mistral-7B-Instruct-v0.1", device_map="auto")

File /usr/local/lib/python3.10/dist-packages/llama_index/llms/huggingface.py:226, in HuggingFaceLLM.__init__(self, context_window, max_new_tokens, query_wrapper_prompt, tokenizer_name, model_name, model, tokenizer, device_map, stopping_ids, tokenizer_kwargs, tokenizer_outputs_to_remove, model_kwargs, generate_kwargs, is_chat_model, callback_manager, system_prompt, messages_to_prompt, completion_to_prompt, pydantic_program_mode, output_parser)
    223 if isinstance(query_wrapper_prompt, str):
    224     query_wrapper_prompt = PromptTemplate(query_wrapper_prompt)
--> 226 self._messages_to_prompt = (
    227     messages_to_prompt or self._tokenizer_messages_to_prompt
    228 )
    230 super().__init__(
    231     context_window=context_window,
    232     max_new_tokens=max_new_tokens,
   (...)
    248     output_parser=output_parser,
    249 )

File /usr/local/lib/python3.10/dist-packages/pydantic/main.py:357, in pydantic.main.BaseModel.__setattr__()

ValueError: "HuggingFaceLLM" object has no field "_messages_to_prompt"
```

**Comment by rudrakhsha-c** [comment]
@joergwa I am facing a similar issue as you. I have tried the same method with different versions of llama-index but yet no success. I have tried using different models too. One of them beingmeta-llama/Llama-2-7b-chat-hfand other wasStabilityAI/stablelm-tuned-alpha-3b. Irrespective of which model I am using, I get the same error. 
I think its an error with initialization of the HuggingFaceLLM object itself and not any other error.

Until yesterday, I could successfully execute the same API with the same models. Must have been some issue with any of the PR's merged within the last 24 hours.
